Only build the test tool and ld-iscsi if we have shared libraries

Backporting an upstream patch to fix a failure when doing a static
build:

/br/output/host/usr/bin/mipsel-ctng-linux-uclibc-gcc -shared -o 
ld_iscsi.so ld_iscsi.o -ldl
/br/output/host/opt/ext-toolchain/bin/../lib/gcc/mipsel-ctng-linux-uclibc/4.8.2/../../../../mipsel-ctng-linux-uclibc/bin/ld: 
ld_iscsi.o: relocation R_MIPS_HI16 against `__gnu_local_gp' can not be 
used when making a shared object; recompile with -fPIC
ld_iscsi.o: could not read symbols: Bad value
collect2: error: ld returned 1 exit status

Upstream commit:
  https://github.com/sahlberg/libiscsi/commit/3d6c2be3428d88682cecfa2a4ad172aaacffb318

Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>

From 3d6c2be3428d88682cecfa2a4ad172aaacffb318 Mon Sep 17 00:00:00 2001
From: Ronnie Sahlberg <ronniesahlberg@gmail.com>
Date: Wed, 22 Oct 2014 17:44:03 -0700
Subject: [PATCH] configure: only build the test tool and ld-iscsi iff we have shared libraries

Signed-off-by: Ronnie Sahlberg <ronniesahlberg@gmail.com>
---
 configure.ac |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5e18ff4..e9c5c93 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,8 +10,6 @@ AM_PROG_AR
 LT_INIT
 
 AC_CANONICAL_HOST
-AM_CONDITIONAL(LD_ISCSI,
-               [expr "$host_os" : linux > /dev/null 2>&1])
 AM_PROG_CC_C_O
 
 enable_write_strings="yes"
@@ -108,8 +106,9 @@ else
   AC_MSG_NOTICE(You need libcunit to build the test suite.)
   AC_MSG_NOTICE(The scsi/iscsi test suite will not be built.)
 fi
-AM_CONDITIONAL(ISCSITEST, [test "$ac_cv_have_cunit" = yes])
+AM_CONDITIONAL(ISCSITEST, [test "$ac_cv_have_cunit" = yes -a "$enable_shared" = "yes"])
 
+AM_CONDITIONAL(LD_ISCSI, [expr "(" "$host_os" : "linux" ")" "&" "$enable_shared" "=" "yes"])
 
 AC_CHECK_MEMBER([struct CU_SuiteInfo.pSetUpFunc],
 		[AC_DEFINE([HAVE_CU_SUITEINFO_PSETUPFUNC], 1,
-- 
1.7.1

